src/Features/Core/Portable/PullMemberUp/AbstractPullMemberUpRefactoringProvider.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp.Dialog;
+using Microsoft.CodeAnalysis.PullMemberUp.QuickAction;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    internal abstract partial class AbstractPullMemberUpRefactoringProvider : CodeRefactoringProvider
+    {
+        protected readonly IPullMemberUpOptionsService _pullMemberUpOptionsService;
+
+        internal AbstractPullMemberUpRefactoringProvider(IPullMemberUpOptionsService pullMemberUpService)
+        {
+            _pullMemberUpOptionsService = pullMemberUpService;
+        }
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
```

Comment:
place document and cancellation token into locals since you keep using them.  it will make all the code downstream much simpler :)

Reply:
same comment applies.